dfastmi/gui/dialog_model.py

```diff
@@ -0,0 +1,126 @@
+        except:
+            if filename != "dfastmi.cfg":
+                return False                
+            return True
+    
+        self.section = self.config["General"]
+
+    def check_configuration(self, branch : Branch, reach: AReach, reference_files: List, measure_files : List) -> bool :
+        config = self.get_configuration(branch, reach, reference_files, measure_files)
+        return check_configuration(self.rivers, config)
+    
+
+    def run_analysis(self) -> bool:
+        # Logic to run analysis based on configuration
+        """
+        Run the D-FAST Morphological Impact analysis based on settings in the GUI.
+
+        Arguments
+        ---------
+        None
+        """
+        try:
+            success = dfastmi.batch.core.batch_mode_core(self.rivers, False, self.config)
+        except:
+            success = False
+        return success
+    
+    
+    def get_configuration(self, branch : Branch, reach: AReach, reference_files: List, measure_files : List) -> ConfigParser:
+        """
+        Extract a configuration from the GUI.
+
+        Arguments
+        ---------
+        None
+
+        Returns
+        -------
+        config : Optional[configparser.ConfigParser]
+            Configuration for the D-FAST Morphological Impact analysis.
+        """
+        config = ConfigParser()
+        config.optionxform = str
+        config.add_section("General")
+        config["General"]["Version"] = "2.0"
+        config["General"]["Branch"] = branch.name
+        config["General"]["Reach"] = reach.name
+        config["General"]["Qthreshold"] = str(reach.qstagnant)
+        config["General"]["Ucrit"] = str(reach.ucritical)
+        config["General"]["OutputDir"] = self.section["OutputDir"] if self.config and self.section and self.config.has_option("General", "OutputDir") and self.section["OutputDir"] else ""
+        config["General"]["Plotting"] = self.section["Plotting"] if self.config and self.section and self.config.has_option("General", "Plotting") and self.section["Plotting"] else "False"
+        config["General"]["SavePlots"] = self.section["SavePlots"] if self.config and self.section and self.config.has_option("General", "SavePlots") and self.section["SavePlots"] else "False"
+        config["General"]["FigureDir"] = self.section["FigureDir"] if self.config and self.section and self.config.has_option("General", "FigureDir") and self.section["FigureDir"] else ""
+        config["General"]["ClosePlots"] = self.section["ClosePlots"] if self.config and self.section and self.config.has_option("General", "ClosePlots") and self.section["ClosePlots"] else "False"
+
+        # Ensure both lists have the same length
+        num_files = min(len(reference_files), len(measure_files))
+        
+        # loop over conditions cond = "C1", "C2", ...        
+        for i, discharge in enumerate(reach.hydro_q[:num_files]): # Ensure it loops until the minimum length
```

Comment:
maybe move this for loop to a seperate descriptive method if possible

Reply:
good idea! will also do with the application settings